### PR TITLE
Tritium Balance + TTV Changes

### PR DESCRIPTION
**Pull request by vincentiusvin**
## About The Pull Request
People have expressed dissatisfaction on how tritium can be abused in overpowered singular tank bombs. This attempts to solve that.

Tritium (and h2) is now made to require 6e7 joules of energy in order to do the higher reaction rate combustion. For reference a 3039kpa 70L Noblium tank contains around 5.1e7 joules. Tritiumfire and h2fire is also slightly rewritten to have a universal reaction rate and be less nuanced.

TTV is now made to inject noblium directly to the gasmix, facilitating the aforementioned energy requirements. In order to do this TTV will now only accept a single payload tank. TTV code has also been slightly refactored into three main procs: log_activation for administration purposes, merge_gases to handle the gas injection and reaction, and calculate_power to return the range. This should hopefully reduce clutter and make it easier to build stuffs with it. This would also mean that you no longer need a hotmix.

Because of how pressure works with noblium, you can make a maxpoints bomb simply by putting a gas with low heat capacity in the payload. To address this, TTV explosions will not scale based on pressure but energy. The reaction tick has also been lowered from two to one in order to make tritium easier to scale. Optimal ratio will be 50/50 with just a tad more oxygen than tritium.  (need more input on this part).

Proto nitrate hydrogen response is now also capped at 20 kelvins so that they cant be used as a fuse for nobliumformation bombs.

Oh and a few minor sprite changes to TTVs too.

## Why It's Good For The Game
Attempts to address a longstanding balance issue regarding singlecaps.

## Changelog
:cl:
balance: Tritium and hydrogen now requires 6e7 joules to react violently.
balance: Proto Nitrate Hydrogen Response is now capped at a minimum of 20 kelvins to occur. 
refactor: TTVs no longer requires a hotmix. TTVs also scale based on energy released from reactions and not pressure. Reaction tick is also lowered from two to one.
/:cl:

### CODE REVIEW

**Comment by Sparkezel** [comment]
Can you show the new sprites bro? Also I never heard of dissatisfaction about single tank maxcaps, other than some salt every so and then but that unavoidable on our saltmines in space.

**Comment by vincentiusvin** [comment]
> Can you show the new sprites bro? Also I never heard of dissatisfaction about single tank maxcaps, other than some salt every so and then but that unavoidable on our saltmines in space.

Certainly!
It's not really new sprites, since i only added some minor details to the inhand file and reused older tank sprite by fiddling with the backend.
image
image
image
image
image

As for the reasoning behind this, several maintainers and admins have expressed how tritium singlecaps are overpowered. I too have used the aforementioned bombs extensively in the past, and it gets old really quick, both for me and for others. The ability to produce virtually a limitless amount of maxcaps is ridiculously overpowered, especially considering the miniscule amount of game knowledge and time investment that it requires.

File: code/__DEFINES/reactions.dm [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #define PLASMA_MINIMUM_OXYGEN_NEEDED		2
 #define PLASMA_MINIMUM_OXYGEN_PLASMA_RATIO	30
 #define FIRE_CARBON_ENERGY_RELEASED			100000	//Amount of heat released per mole of burnt carbon into the tile
-#define FIRE_HYDROGEN_ENERGY_RELEASED		280000  //Amount of heat released per mole of burnt hydrogen and/or tritium(hydrogen isotope)
+#define FIRE_HYDROGEN_ENERGY_RELEASED		2800000  //Amount of heat released per mole of burnt hydrogen and/or tritium(hydrogen isotope)
```
Comment (Sparkezel) [inline code/__DEFINES/reactions.dm:10]: I think you should document that

Comment (vincentiusvin) [reply code/__DEFINES/reactions.dm:10]: will do

**Comment by interestingusernam3** [comment]
>As for the reasoning behind this, several maintainers and admins have expressed how tritium singlecaps are overpowered. I too have used the aforementioned bombs extensively in the past, and it gets old really quick, both for me and for others. The ability to produce virtually a limitless amount of maxcaps is ridiculously overpowered, especially considering the miniscule amount of game knowledge and time investment that it requires.

And because of that you decided to fix it by making people set up a plasma burner and watch paint dry to make the biggest bomb possible?

**Comment by vincentiusvin** [comment]
> And because of that you decided to fix it by making people set up a plasma burner and watch paint dry to make the biggest bomb possible?

Good point! 
I would love it if tritium production (which i should remind us, is a factor both in toxins AND also singlecaps) is more fleshed out. Or even better, that more interesting and novel methods to produce bombs are added to the game. If you have any ideas/suggestions/PRs, feel free to let everyone know!

**Comment by interestingusernam3** [comment]
> I would love it if tritium production (which i should remind us, is a factor both in toxins AND also singlecaps) is more fleshed out. Or even better, that more interesting and novel methods to produce bombs are added to the game. If you have any ideas/suggestions/PRs, feel free to let everyone know!

This problem isn't severe enough to warrant a rushed and terrible solution.

**Review by Twaticus — APPROVED** [review]

File: code/game/objects/effects/spawners/bombspawner.dm [hunk 2]
```diff
@@ -1,36 +1,25 @@
...
-	OT.air_contents.gases[/datum/gas/oxygen][MOLES] = pressure_o*OT.volume/(R_IDEAL_GAS_EQUATION*CELSIUS_TO_KELVIN(temp_o))
-	OT.air_contents.temperature = CELSIUS_TO_KELVIN(temp_o)
-
-	V.tank_one = PT
-	V.tank_two = OT
+	PT.air_contents.assert_gas(/datum/gas/tritium)
+	PT.air_contents.assert_gas(/datum/gas/oxygen)
+	PT.air_contents.temperature = T20C
+	PT.air_contents.gases[/datum/gas/tritium][MOLES] = 2
+	PT.air_contents.gases[/datum/gas/oxygen][MOLES] = 3
```
Comment (LemonInTheDark) [inline code/game/objects/effects/spawners/bombspawner.dm:19]: does this maxcap currently?

Comment (vincentiusvin) [reply code/game/objects/effects/spawners/bombspawner.dm:19]: Havent tested it yet, but 2*2.8e6 is 5.6e6, which is around 37 range if the math checks out. So yes it should.

**Comment by vincentiusvin** [comment]
Many disagreed with the approach, will revisit the idea on a later date.

**Pull request closed without merging**